tests/generator/test_list_to_batches.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+from chia.util.generator_tools import list_to_batches
+
+
+def test_empty_lists():
+    for remaining, batch in list_to_batches([], 1):
+        assert remaining == 0
+        assert batch == []
```

Comment:
I'm a bit surprised that the loop body is invoked at all on an empty list. Is this deliberate?

Reply:
if it is indeed supposed to run the body once here (assuming there's a good reason), I think it would be good to have an assert making sure it's is called exactly once too, If `list_to_batches()` is changed to not run the body on an empty list, this test would still pass as it's written right now

Reply:
Good point, there is no good reason, feels more reasonable to just return an empty iterator in case of an empty list, see fa7b5b1a837fa57e1f3bcccec4884e5f772a32fa.